text/0456-elasticache-l2.md

```diff
@@ -0,0 +1,375 @@
+# Amazon ElastiCache L2
+
+* **Original Author(s):**: @dbartholomae
+* **Tracking Issue**: #456
+* **API Bar Raiser**: @corymhall
+
+The `aws-elasticache` construct library allows you to create Amazon ElastiCache
+data stores with just a few lines of code. As with most construct libraries,
+you can also easily define permissions and metrics using a simple API.
+
+The amount of effort needed to create these resources is about the same as doing
+it using the AWS console, sometimes even less due to easier set up of VPC connectivity.
+
+## Working Backwards
+
+### CHANGELOG
+
+`feat(elasticache): RedisReplicationGroup L2`
+
+### README
+
+---
+
+# Amazon ElastiCache Construct Library
+
+[Amazon ElastiCache](https://aws.amazon.com/elasticache/) is a fully managed,
+in-memory caching service supporting flexible, real-time use cases. You can use
+ElastiCache for caching, which accelerates application and database performance,
+or as a primary data store for use cases that don't require durability like
+session stores, gaming leaderboards, streaming, and analytics. ElastiCache is
+compatible with Redis and Memcached.
+
+This module is part of the [AWS Cloud Development Kit](https://github.com/aws/aws-cdk)
+project. It allows you to define ElastiCache Redis replication groups.
+
+## Constructs
+
+### Defining a RedisReplicationGroup
+
+In order to define a `RedisReplicationGroup` (Redis (cluster mode disabled)),
```

Comment:
They accept different props, so separating them will make it more obvious which settings are required for each.

Reply:
If a tenet of the L2 constructs is to codify best practices, I would actually argue we should only support cluster mode enabled. I would suggest the default be 3 shards with 1 replica. Having a single shard has lower availability than 3 shards.

Reply:
What would be the cost effect? Is it reasonable to not use cluster mode for e.g. small companies which want to save money and don't mind occasionally being offline?

Reply:
Yes, some folks may prefer the smaller deployment for cost saving reasons at the cost of some availability. We've had various discussions about optimizing this case for better availability, so it might change it the future.

Reply:
I definitely think that we should have two constructs. It's fine if we just start with one or the other, but I don't want a single one that works with both.

Reply:
I'll resolve this conversation then without a change to the RFC.